src/commons/controlBar/ControlBarShareButton.tsx

```diff
@@ -135,4 +158,31 @@ export class ControlBarShareButton extends React.PureComponent<ControlBarShareBu
       this.shareInputElem.current.select();
     }
   }
+
+  private fetchUUID() {
+    const requestBody = {
+      shared_program: {
+        data: this.props.programConfig
+      }
+    };
+    const fetchOpts: RequestInit = {
+      method: 'POST',
+      body: JSON.stringify(requestBody),
+      headers: {
+        'Content-Type': 'application/json'
+      }
+    };
+    fetch(`${Constants.backendUrl}/api/shared_programs`, fetchOpts)
+      .then(res => {
+        return res.json();
+      })
+      .then(resp => {
+        this.setState({
+          // seems like there's no frontend url env variable, should be replaced by frontend server accordingly
+          keyword: `http://localhost:8000/playground/share/` + resp.uuid
```

Comment:
CRA injects an environment variable called `PUBLIC_URL`, no hardcoding is needed.

Reply:
I found out `env.process.PUBLIC_URL` is undefined in actual running, so I replaced it as `window.location.host` here.